stdlib/public/core/Sort.swift

```diff
@@ -337,6 +337,7 @@ extension MutableCollection where Self: BidirectionalCollection {
 /// - Precondition: `buffer` must point to a region of memory at least as large
 ///   as `min(mid - lo, hi - mid)`.
 /// - Postcondition: `lo..<hi` is sorted according to `areInIncreasingOrder`.
+@discardableResult
```

Comment:
@glessard I'm sorry, I forgot to ping you to review. I added `@discardableResult` to `_merge` as @LucianoPAlmeida suggested.